chrome/background.js

```diff
@@ -1,16 +1,134 @@
-let languages = ['english', 'spanish', 'french', 'portuguese', 'russian']
+let glossaries = {
+	'english': 'english-terminology-for-editors',
+	'spanish': 'spanish-glossary',
+	'french': 'french-glossary',
+	'portuguese': 'portuguese-glossary',
+	'russian': 'russian-glossary'
+}
 
-chrome.runtime.onInstalled.addListener(function() {
+chrome.runtime.onInstalled.addListener(init)
+chrome.runtime.onStartup.addListener(init)
 
-	for (let i in languages) {
-		let lang = languages[i]
-		let glossaryName = lang + '-glossary'
-		const url = chrome.runtime.getURL('data/' + glossaryName + '.json');
+
+function init() {
+	getGlossaries().catch(() => {
+		console.log('Could not get ForumBee. Using local glossary instead.')
+		for (let lang in glossaries) {
+			let glossaryName = lang + '-glossary'
+			const url = chrome.runtime.getURL('data/' + glossaryName + '.json');
+
+			fetch(url).then((resp) => {
+				return resp.json()
+			}).then((data) => {
+				let obj = {}
+				obj[glossaryName] = data
+				chrome.storage.local.set(obj, function() {
+			        console.log(lang + ' glossary set!');
+			    });
+			})
+		}
+	})
+}
+
+async function getGlossaries() {
+	for (let lang in glossaries) {
+		console.log('starting ' + lang)
+		let g = glossaries[lang]
+		let data = []
+		let i = 0
+		let isLast = false
+		while (!isLast) {
+			let url = `https://kiva.forumbee.com/api/2/posts?fields=postKey,title,text&categoryLink=${g}&postType=topics&textFormat=plain&limit=1000&offset=${i*1000}`
+			let site = await fetch(url, { credentials: 'include' })
+			let page = await site.json()
+
+			if (page.data.length < 1000) {
+				isLast = true
+			}
+
+			if (haveCommonEntries(data, page.data)) {
+				break;
+			}
+
+			data = data.concat(page.data)
+
+			i++
+		}
+
+		let glossary = await parseGlossary(data)
 
 		let obj = {}
-		obj[glossaryName] = url
-		chrome.storage.sync.set(obj, function() {
+		obj[lang + '-glossary'] = glossary
+		chrome.storage.local.set(obj, function() {
 	        console.log(lang + ' glossary set!');
 	    });
 	}
-});
\ No newline at end of file
+}
+
+let re = /( [^ a-zA-Z0-9\-]| or|\/)/
+let countryUrl = 'https://gist.githubusercontent.com/kalinchernev/486393efcca01623b18d/raw/daa24c9fea66afb7d68f8d69f0c4b8eeb9406e83/countries'
```

Comment:
@kevinzwang Is the advantage here of a url that it might get updated over time? Otherwise, perhaps we should just export the country list and add it to the repo?

Reply:
@kevinzwang or rather, looking at what you have below, why use the URL here but the countries.txt list below?

Reply:
Actually, we should just use the local file